util/app/discovery/discovery.go

```diff
@@ -62,6 +64,32 @@ func Discover(ctx context.Context, appPath, repoPath string, enableGenerateManif
 		if kustomize.IsKustomization(base) && IsManifestGenerationEnabled(v1alpha1.ApplicationSourceTypeKustomize, enableGenerateManifests) {
 			apps[dir] = string(v1alpha1.ApplicationSourceTypeKustomize)
 		}
+
+		if (strings.HasSuffix(base, ".yaml") || strings.HasSuffix(base, ".yml")) && IsManifestGenerationEnabled(v1alpha1.ApplicationSourceTypeDirectory, enableGenerateManifests) {
+			f, err := os.Open(path)
+			if err != nil {
+				log.Errorf("failed to open %s: %v", path, err)
```

Comment:
Probably need to dump out of directory detection at this point. Maybe pull this block into a function so you can just `return` out?

Reply:
Yeah makes sense the block was getting too big anyway